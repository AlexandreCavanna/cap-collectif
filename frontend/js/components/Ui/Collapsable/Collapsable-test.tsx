/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Collapsable from '~ui/Collapsable'

describe('<Collapsable />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Collapsable>
        <Collapsable.Button>
          <p>Tri</p>
        </Collapsable.Button>
        <Collapsable.Element ariaLabel="Contenu">
          <h1>Je suis caché hihi faut cliquer sur Tri pour me voir</h1>
        </Collapsable.Element>
      </Collapsable>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly when aligned to right', () => {
    const wrapper = shallow(
      <Collapsable align="right">
        <Collapsable.Button>
          <p>Tri</p>
        </Collapsable.Button>
        <Collapsable.Element ariaLabel="Contenu">
          <h1>Je suis caché hihi faut cliquer sur Tri pour me voir</h1>
        </Collapsable.Element>
      </Collapsable>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly when no aria-label provided', () => {
    const wrapper = shallow(
      <Collapsable align="right">
        <Collapsable.Button>
          <p>Tri</p>
        </Collapsable.Button>
        <Collapsable.Element>
          <h1>Je suis caché hihi faut cliquer sur Tri pour me voir</h1>
        </Collapsable.Element>
      </Collapsable>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should render correctly when disabled', () => {
    const wrapper = shallow(
      <Collapsable disabled>
        <Collapsable.Button>
          <p>Tri</p>
        </Collapsable.Button>
        <Collapsable.Element ariaLabel="Contenu">
          <h1>Je suis caché hihi faut cliquer sur Tri pour me voir</h1>
        </Collapsable.Element>
      </Collapsable>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
