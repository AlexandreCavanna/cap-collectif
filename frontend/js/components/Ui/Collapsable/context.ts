import * as React from 'react'
export type Context = {
  readonly visible: boolean
  readonly setVisible: (value: boolean | ((updaterValue: boolean) => boolean)) => void
  readonly onClose?: () => void | Promise<any>
  readonly disabled?: boolean
}
export const CollapsableContext = React.createContext<Context>({
  visible: false,
  setVisible: () => undefined,
  onClose: () => undefined,
  disabled: false,
})
