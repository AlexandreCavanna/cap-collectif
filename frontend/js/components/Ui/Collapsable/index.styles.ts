import styled, { css } from 'styled-components'
import { slideInUp } from '~/utils/styles/keyframes'
import type { CollapsableAlignment } from '~ui/Collapsable/index'

export const CollapsableBody = styled.div<{
  isAbsolute: boolean
}>`
  animation: ${slideInUp} 0.2s forwards;
  ${props =>
    props.isAbsolute &&
    css`
      position: absolute;
      top: calc(100% + 10px);
      z-index: 100;
    `}
`
export const Container = styled.div<{
  align: CollapsableAlignment
  disabled?: boolean
}>`
  display: inline-block;
  position: relative;
  ${props =>
    props.align === 'left' &&
    css`
      &
        ${
          /* sc-selector */
          CollapsableBody
        } {
        left: 0;
      }
    `}
  ${props =>
    props.align === 'right' &&
    css`
      &
        ${
          /* sc-selector */
          CollapsableBody
        } {
        right: 0;
      }
    `}
  ${props =>
    props.disabled &&
    css`
      opacity: 0.5;
      border: 1px solid #eee;
      background-color: #fff;
    `}
`
export const Button = styled.div<{
  visible: boolean
  inline: boolean
}>`
  display: ${props => (props.inline ? 'inline-flex' : 'flex')};
  align-items: center;

  > * {
    margin: 0;
  }

  & svg {
    transition: transform 0.2s;
    margin-left: 4px;
  }

  ${props =>
    props.visible &&
    css`
      & svg {
        transform: rotate(-180deg);
      }
    `}

  &:hover {
    cursor: pointer;
  }
`
