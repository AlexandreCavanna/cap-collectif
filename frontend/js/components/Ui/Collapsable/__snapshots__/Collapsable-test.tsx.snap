// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsable /> should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "disabled": undefined,
      "onClose": undefined,
      "setVisible": [Function],
      "visible": false,
    }
  }
>
  <indexstyles__Container
    align="left"
  >
    <CollapsableButton>
      <p>
        Tri
      </p>
    </CollapsableButton>
    <CollapsableElement
      ariaLabel="Contenu"
    >
      <h1>
        Je suis caché hihi faut cliquer sur Tri pour me voir
      </h1>
    </CollapsableElement>
  </indexstyles__Container>
</ContextProvider>
`;

exports[`<Collapsable /> should render correctly when aligned to right 1`] = `
<ContextProvider
  value={
    Object {
      "disabled": undefined,
      "onClose": undefined,
      "setVisible": [Function],
      "visible": false,
    }
  }
>
  <indexstyles__Container
    align="right"
  >
    <CollapsableButton>
      <p>
        Tri
      </p>
    </CollapsableButton>
    <CollapsableElement
      ariaLabel="Contenu"
    >
      <h1>
        Je suis caché hihi faut cliquer sur Tri pour me voir
      </h1>
    </CollapsableElement>
  </indexstyles__Container>
</ContextProvider>
`;

exports[`<Collapsable /> should render correctly when disabled 1`] = `
<ContextProvider
  value={
    Object {
      "disabled": true,
      "onClose": undefined,
      "setVisible": [Function],
      "visible": false,
    }
  }
>
  <indexstyles__Container
    align="left"
    disabled={true}
  >
    <CollapsableButton>
      <p>
        Tri
      </p>
    </CollapsableButton>
    <CollapsableElement
      ariaLabel="Contenu"
    >
      <h1>
        Je suis caché hihi faut cliquer sur Tri pour me voir
      </h1>
    </CollapsableElement>
  </indexstyles__Container>
</ContextProvider>
`;

exports[`<Collapsable /> should render correctly when no aria-label provided 1`] = `
<ContextProvider
  value={
    Object {
      "disabled": undefined,
      "onClose": undefined,
      "setVisible": [Function],
      "visible": false,
    }
  }
>
  <indexstyles__Container
    align="right"
  >
    <CollapsableButton>
      <p>
        Tri
      </p>
    </CollapsableButton>
    <CollapsableElement>
      <h1>
        Je suis caché hihi faut cliquer sur Tri pour me voir
      </h1>
    </CollapsableElement>
  </indexstyles__Container>
</ContextProvider>
`;
